ansible/roles/lib_aos_modules/library/aos_rhsm_repos.py

```diff
@@ -0,0 +1,211 @@
+                              'name': repo[1].split('Name:')[1].strip(),
+                              'url': repo[2].split('URL:')[1].strip(),
+                              'enabled': int(repo[3].split('Enabled:')[1].strip()) == 1
+                             })
+
+        return all_repos
+
+
+    def query_repos(self, qtype='list'):
+        """query for repos"""
+
+        cmd = [RHSM_BIN, 'repos']
+
+        if qtype == 'enabled':
+            cmd.append('--list-enabled')
+        elif qtype == 'disabled':
+            cmd.append('--list-disabled')
+        else:
+            cmd.append('--list')
+
+        rcode, out, err = self.module.run_command(cmd)
+        if rcode != 0:
+            self.module.fail_json(msg={'rc': rcode, 'cmd': cmd,
+                                       'err': 'Error querying repositories: [%s]' % err,
+                                      })
+
+        if 'no available repositories matching' in out:
+            return []
+
+        try:
+            return RHSMRepos.parse_repos(out)
+        except RHSMError as err:
+            self.module.fail_json(msg=err)
+
+    def needs_update(self):
+        """verify that the enabled repos are enabled"""
+        repos = self.query_repos(qtype='enabled')
+        if len(repos) != len(self.module.params['enabled']):
+            return True, repos
+
+        # build sets of the repo ids and compare them
+        rids = set([rep['id'] for rep in repos])
+        ren = set(self.enabled)
+        if rids - ren or ren - rids:
+            return True, repos
+
+        return False, repos
+
+    def check_identity(self):
+        """verify that this system is entitled"""
+        rcode, _, _ = self.module.run_command([RHSM_BIN, 'identity'])
+        if rcode != 0:
+            self.module.fail_json(msg="command `subscription-manager identity` failed.  exit code non zero.")
+
+    @staticmethod
+    def run_ansible(module):
+        """run the ansible code"""
+        rhsm = RHSMRepos(module)
+
+        # step 1: verify we are entitled
+        rhsm.check_identity()
+
+        # step 2: if we are state=list, query and return results
+        if rhsm.module.params['state'] == 'list':
+            repos = rhsm.query_repos(qtype=module.params['query'])
+
+            rhsm.module.exit_json(changed=False, repos=repos)
+
+        # step 2: if we are state=present, check if we need to update
+        elif module.params['state'] == 'present':
+            update, repos = rhsm.needs_update()
+            if not update:
+                module.exit_json(changed=update, repos=repos)
+
+            # exit if check mode before we make changes
+            if module.check_mode:
+                module.exit_json(changed=False, repos=repos)
+
+            # step 3: we need to set our repositories
+            rhsm.set_repositories()
+            repos = rhsm.query_repos(qtype='enabled')
+            module.exit_json(changed=update, repos=repos)
```

Comment:
Same here as above, we know that it changed because we wouldn't be running this code if it didn't, so I prefer `changed=True` just to be explicit for the next person reading the code.